src/opf/ed.jl

```diff
@@ -0,0 +1,500 @@
+        model.ext[:termination_info] = Dict{Symbol,Any}(
+            :primal_status => MOI.FEASIBLE_POINT,
+            :dual_status => MOI.FEASIBLE_POINT,
+        )
+    else
+        model.ext[:termination_info] = Dict{Symbol,Any}(
+            :primal_status => MOI.UNKNOWN_RESULT_STATUS,
+            :dual_status => MOI.UNKNOWN_RESULT_STATUS,
+        )
+    end
+
+    model.ext[:termination_info][:solve_time] = solve_time
+    model.ext[:termination_info][:ptdf_iterations] = niter
+    model.ext[:termination_info][:termination_status] = st
+
+    return
+end
+
+function extract_result(opf::OPFModel{EconomicDispatch})
+    data = opf.data
+    model = opf.model
+
+    ref = PM.build_ref(data)[:it][:pm][:nw][0]
+    N = length(ref[:bus])
+    G = length(ref[:gen])
+    E = length(data["branch"])
+
+    res = Dict{String,Any}()
+    res["opf_model"] = string(model.ext[:opf_model])
+    res["objective"] = JuMP.objective_value(model)
+    res["objective_lb"] = -Inf
+    res["optimizer"] = JuMP.solver_name(model)
+
+    res["solve_metadata"] = model.ext[:solve_metadata]
+    if model.ext[:solve_metadata][:iterative_ptdf]
+        tinfo = model.ext[:termination_info]
+        res["termination_status"] = tinfo[:termination_status]
+        res["primal_status"] = tinfo[:primal_status]
+        res["dual_status"] = tinfo[:dual_status]
+        res["solve_time"] = tinfo[:solve_time]
+        res["ptdf_iterations"] = tinfo[:ptdf_iterations]
+    else
+        res["termination_status"] = JuMP.termination_status(model)
+        res["primal_status"] = JuMP.primal_status(model)
+        res["dual_status"] = JuMP.dual_status(model)
+        res["solve_time"] = JuMP.solve_time(model)
+    end
+    res["minimum_reserve"] = get(data, "minimum_reserve", 0.0)
+
+    res["solution"] = sol = Dict{String,Any}()
+
+    sol["per_unit"] = get(data, "per_unit", false)
+    sol["baseMVA"]  = get(data, "baseMVA", 100.0)
+
+    sol["gen"] = Dict{String,Any}()
+    for g in 1:G
+        sol["gen"]["$g"] = Dict(
+            "pg" => value(model[:pg][g]),
+            "r" => value(model[:r][g]),
+            "rmin" => get(data["gen"]["$g"], "rmin", 0.0),
+            "rmax" => get(data["gen"]["$g"], "rmax", 0.0),
+
+            "mu_pg" => dual(LowerBoundRef(model[:pg][g])) - dual(UpperBoundRef(model[:pg][g])),
+            "mu_r" => dual(LowerBoundRef(model[:r][g])) - dual(UpperBoundRef(model[:r][g])),
+
+            "mu_total_generation" => dual(model[:total_generation][g]),
+        )
+    end
+
+    sol["branch"] = Dict{String,Any}()
+    for e in 1:E
+        if data["branch"]["$e"]["br_status"] == 0
+            sol["branch"]["$e"] = Dict(
+                "pf" => 0,
+                "df" => 0,
+                "mu_pf" => 0,
+                "mu_df" => 0,
+                "lam_ptdf" => 0,
+            )
+        else
+            sol["branch"]["$e"] = Dict(
+                "pf" => value(model[:pf][e]),
+                "df" => value(model[:δf][e]),
```

Comment:
Noted by @klamike: it would be safer to re-compute the power flows rather than export the solver solution. Non-tracked branches may have garbage values here.

Another option is to document this behavior, and be very clear about what we return. I think the previous option is more fool-proof though.